src/helpers/cosmiconfig.ts

```diff
@@ -40,9 +40,24 @@ export function createCosmiConfigSync(moduleName: string, legacy: boolean) {
 }
 
 const loadTypeScript: Loader = (...args) => {
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { TypeScriptLoader } = require('cosmiconfig-typescript-loader');
+    return TypeScriptLoader({ transpileOnly: true })(...args);
+  } catch (err) {
+    if (isRequireESMError(err)) {
+      return jitiLoader(...args);
+    }
+    throw err;
+  }
+};
+
+const jitiLoader: Loader = (filepath) => {
   // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { TypeScriptLoader } = require('cosmiconfig-typescript-loader');
-  return TypeScriptLoader({ transpileOnly: true })(...args);
+  const jiti = require('jiti')(__filename, {
```

Comment:
I guess you can use `import` statement instead (since this package in `"dependencies"`), `cosmiconfig-typescript-loader` used with `require` because it's a peerDep with peerDependecyMeta